lib/url.js

```diff
@@ -192,6 +194,18 @@ Url.prototype.parse = function(url, parseQueryString, slashesDenoteHost) {
     if (atSign !== -1) {
       auth = rest.slice(0, atSign);
       rest = rest.slice(atSign + 1);
+
+      var username, password;
+      colon = auth.lastIndexOf(':');
+      if (colon !== -1) {
+        username = auth.slice(0, colon);
```

Comment:
I agree the point of doing it like the browser. But it shouldn't matter if you split on the first or the last, because there should only ever be one. It's invalid to have a encoded colon in the username or password. They should be encoded.